comp/core/tagger/setup_fake.go

```diff
@@ -0,0 +1,49 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build test
+// +build test
+
+package tagger
+
+import (
+	"context"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/comp/core"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/local"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// GetDefaultTagger returns the default Tagger in gvien instance
+// For testing only, defaultTagger should not be used directly in production
+func GetDefaultTagger(taggerClient Component) Component {
```

Comment:
If you need to have a specific method for testing purpose only, you can add this method to `MockComponent`.